text/0000-wallet-replay-protection.md

```diff
@@ -0,0 +1,322 @@
+Unsafe receives are allowed, but safe receives (that are necessarily payjoins) are preferred as long as the receiver gets to finalize, which minimizes the risk of utxo spoofing. Sends should always be a safe transaction and thus _require_ spending an output from an earlier safe transaction.
+
+A safe cancel requires an immediate self-spend of an input of the tx to be canceled.
+
+The wallet rules for spending safely would happen behind the scenes, so the end user would not be aware of any changes. Canceling a transaction would be a difference experience than it is now, but it is required to change it to prevent the play attacks from occuring. A possible cancel flow would be to make the cancellation a longer event which is not finished until the self-spend transaction has been confirmed on the chain.  
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## General technical overview
+
+As we have already mentioned, a transaction can be replayed if the inputs are recreated and its outputs don't exist. One way to stop this attack would be to make it impossible to recreate an input. Of course the owner can always recreate it, but the problem is really if anyone else recreates it by replaying the transaction that created the input. Luckily, there is a way to make an input that is protected and can't be recreated by anyone else except by the owner. This seems impossible because you can always just go one transaction back and recreate the outputs that created the inputs and keep repeating this until you hit a transaction that can be replay at which point you can replay all the next transactions and thus successfully recreate the target input. To prevent this, we create a 'stopping' transaction `T` for which we _know_ it can't be recreated. We achieve this by creating an output in `T` that we call an `anchor` output which will never be spent. Since Grin does not allow duplicate outputs, it means that the transaction that created `anchor` output can't be recreated because it would attempt to insert a duplicate output in the UTXO set which is invalid by Grin rules. Along with an `anchor` output we create additional outputs `O1` and `O2` in `T` which we call `Protected` outputs - it's not obvious why they have this name yet.
+
+Let's say we have a transaction `T`
+```
+I1  -> anchor
+    -> O1
+    -> O2
+```
+
+We now use `O2` in a new transaction `T2` as an input to send money to Bob.
+
+```
+O2  -> O3 (change output)
+    -> Bob_output
+```
+
+The key idea here is to observe that `T2` can't be replayed by anyone else. Only we can create the input needed for the transaction to be replayed `O2`. The attacker could attempt to replay the previous transaction `T` to create the  `O2` input, but `T` can't be replayed because it has an `anchor` output that has not been spent and hence can't be valid. This means that our transaction `T2` is protected against other parties replaying it. 

```

Comment:
Reads better as "create the input `O2` needed for the transaction to be replayed."